### PR TITLE
SECURITY-2623:Missing permission check

### PR DESCRIPTION
**Pull request by aasat**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did - compuware -: topaz-for-total-test - Missing permission check
- [x] Link to relevant issues in GitHub or Jira -: https://issues.jenkins.io/browse/SECURITY-2623
- [] Link to relevant pull requests, esp. upstream and downstream changes
- [] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/com/compuware/jenkins/totaltest/TotalTestCTBuilder.java [hunk 1]
```diff
@@ -1585,8 +1585,14 @@ public FormValidation doCheckCredentialsId(@QueryParameter final String value)
...
 		 * @return validation message
 		 */
-		public FormValidation doCheckReportFolder(@QueryParameter final String value)
+		public FormValidation doCheckReportFolder(@QueryParameter final String value, @AncestorInPath Item project)
 		{
+			if (project == null) {
+				Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+			} else {
+				project.checkPermission(Item.CONFIGURE);
+			}
```
Comment (daniel-beck) [inline src/main/java/com/compuware/jenkins/totaltest/TotalTestCTBuilder.java:1594]: Information about the controller file system should be limited to users with Overall/Administer permission.

I recommed either returning `FormValidation#ok` if that permission is missing, possibly with a message telling users that form validation for this field is only available to users with Overall/Administer permission.

The same applies to the other code location.

Comment (aasat) [reply src/main/java/com/compuware/jenkins/totaltest/TotalTestCTBuilder.java]: Hello @daniel-beck,

As per you recommendation, I made changes, please review for same. Please let us know your review points, if any

Thanks,
Ajay

Comment (aasat) [reply src/main/java/com/compuware/jenkins/totaltest/TotalTestCTBuilder.java]: Hello @daniel-beck,
Appreciate your quick response and help.
Could you please provide more details about the issue or possible solution and it location. 
Please let us know if there are any other issues, so we can work on it.
Thanks,
Ajay

Comment (Kevin-CB) [reply src/main/java/com/compuware/jenkins/totaltest/TotalTestCTBuilder.java]: Hello @aasat, The information about the path you want to give are only from the controller file system, they are quite sensitive and must be available only to the administrator of your instance, so it is better not to give any information at all.

Here, Daniel suggests you to only look at the `Jenkins.Administer` permission ,which means to remove your check on `Item.Configure` or even on the `project` and to replace it only by the example he provided you.

Comment (aasat) [reply src/main/java/com/compuware/jenkins/totaltest/TotalTestCTBuilder.java]: Hello @daniel-beck, @Kevin-CB,
Thanks for your quick response.
As suggested, updated code

Thanks,
Ajay

**Review by Kevin-CB — APPROVED** [review]
Perfect! You can already merge it and we'll work on the removal of your warnings

**Comment by Kevin-CB** [comment]
Thanks for your reactivity @aasat, but we'll need a new release to be able to adapt the [PR-666](https://github.com/jenkins-infra/update-center2/pull/666)

**Pull request merged**